### DISCUSSION TITLE
Send email notifications with O365 SMTP smtp.office365.com PORT 587

### DISCUSSION THREAD

**Discussion by skatboy** [containrrr/watchtower#1862] [Q&A]
Hello,

I've been using Watchtower in Synology Docker for several months now and it works really well.
I would now like to configure email notifications.
I've read the documentation about configuring notifications.

My mail server is O365.

My smtp configuration works in Bitwarden, Unifi which are used under Synology Docker.

For Watchtower I have tried to configure notifications as follows:

```
WATCHTOWER_NOTIFICATIONS=email
WATCHTOWER_NOTIFICATION_EMAIL_FROM=email
WATCHTOWER_NOTIFICATION_EMAIL_TO=email
WATCHTOWER_NOTIFICATION_EMAIL_SERVER=smtp.office365.com
WATCHTOWER_NOTIFICATION_EMAIL_SERVER_TLS_SKIP_VERIFY= test yes and no, don’t change anything
WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PORT=587
WATCHTOWER_NOTIFICATION_EMAIL_SERVER_USER=email
WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PASSWORD=app_password
WATCHTOWER_NOTIFICATION_EMAIL_DELAY=2
```

or like this 

`smtp://mail:password@smtp.office365.com:587/?from=mail&to=mail&encryption=ExplicitTLS&Auth=Unknown&UseStartTLS=yes`

depending on the type of encryption or authentication, I get the following error messages :

`time="2023-11-28T01:49:16+01:00" level=error msg="Failed to send shoutrrr notification" error="failed to send using smtp: timed out" index=0 notify=no service=smtp
`

`time="2023-11-28T01:04:33+01:00" level=error msg="Failed to send shoutrrr notification" error="error authenticating: unencrypted connection" index=0 notify=no service=smtp`

`time="2023-11-28T01:51:57+01:00" level=error msg="Failed to send shoutrrr notification" error="error sending message to recipient: error creating new message: 530 5.7.57 Client not authenticated to send mail. [AS4P251CA0018.EURP251.PROD.OUTLOOK.COM 2023-11-28T00:51:57.982Z 08DBEEF16DDB65F8]" index=0 notify=no service=smtp`

The result is that nothing works. I've never had so much trouble configuring email notifications.

I've read that other people have had problems configuring mail notifications via O365.

Perhaps solutions have been found since then?

**Comment by piksel** [comment]
No, Microsofts SMTP service and the go SMTP client does not share compatible authentication implementations. I wrote a custom implementation of LOGIN auth that should allow this to work, but it hasn't been merged into shoutrrr yet.

**Comment by skatboy** [comment]
Ok.

This confirms what I had read.

Do you think this will be implemented quickly or am I better off using another smtp service ?

Thank you for your feedback.
